netspresso/benchmarker/v2/benchmarker.py

```diff
@@ -0,0 +1,199 @@
+            folder_path = Path(input_model_path).parent
+
+            benchmarker_metadata = BenchmarkerMetadata()
+            metadatas = []
+
+            if FileHandler.check_exists(folder_path / f"{file_name}.json"):
+                metadatas = MetadataHandler.load_json(folder_path / f"{file_name}.json")
+                # metadatas.append(asdict(benchmarker_metadata))
+
+            current_credit = auth_client.get_credit(
+                access_token=self.token_handler.tokens.access_token,
+                verify_ssl=self.token_handler.verify_ssl,
+            )
+            check_credit_balance(
+                user_credit=current_credit, service_credit=ServiceCredit.MODEL_BENCHMARK
+            )
+
+            # GET presigned_model_upload_url
+            presigned_url_response = (
+                launcher_client_v2.benchmarker.presigned_model_upload_url(
+                    access_token=self.token_handler.tokens.access_token,
+                    input_model_path=input_model_path,
+                )
+            )
+
+            # UPLOAD model_file
+            launcher_client_v2.benchmarker.upload_model_file(
+                access_token=self.token_handler.tokens.access_token,
+                input_model_path=input_model_path,
+                presigned_upload_url=presigned_url_response.data.presigned_upload_url,
+            )
+
+            # VALIDATE model_file
+            validate_model_response = (
+                launcher_client_v2.benchmarker.validate_model_file(
+                    access_token=self.token_handler.tokens.access_token,
+                    input_model_path=input_model_path,
+                    ai_model_id=presigned_url_response.data.ai_model_id,
+                )
+            )
+
+            # START convert task
+            response = launcher_client_v2.benchmarker.start_task(
+                access_token=self.token_handler.tokens.access_token,
+                input_model_id=presigned_url_response.data.ai_model_id,
+                data_type=validate_model_response.data.detail.data_type,
+                target_device_name=target_device_name,
+                hardware_type=target_hardware_type,
+                input_layer=validate_model_response.data.detail.input_layer,
+                software_version=target_software_version,
+            )
+
+            if wait_until_done:
+                while True:
+                    # Poll Benchmark Task status
+                    response = launcher_client_v2.benchmarker.read_task(
+                        access_token=self.token_handler.tokens.access_token,
+                        task_id=response.data.benchmark_task_id,
+                    )
+                    if response.data.status in [
+                        TaskStatusForDisplay.FINISHED,
+                        TaskStatusForDisplay.ERROR,
+                    ]:
+                        break
+                    time.sleep(3)
+
+            if launcher_client_v2.is_cloud():
```

Comment:
p1.
In case of benchmark failed, credit is not consumed, how about this?
```
if launcher_client_v2.is_cloud() and response.data.status == TaskStatusForDisplay.FINISHED:
...
if response.data.status in [TaskStatusForDisplay.ERROR, TaskStatusForDisplay.TIMEOUT]:
logger.info(bla}"
                )
```

Reply:
Thanks for your feedback. I’ve updated the code as per your suggestions. 👍